dn.go

```diff
@@ -143,6 +143,9 @@ func ParseDN(str string) (*DN, error) {
 			}
 		} else if char == ',' || char == '+' {
 			// We're done with this RDN or value, push it
+			if len(attribute.Type) == 0 {
+				return nil, errors.New("incomplete type, value pair")
```

Comment:
That's caught here or with the "DN ended with incomplete type, value pair" error, adding 
`"=test,DC=example,DC=com": "incomplete type, value pair",` in `dn_test.go:TestErrorDNParsing()` passes.